python/torch_mlir_e2e_test/test_suite/scalar.py

```diff
@@ -151,3 +151,123 @@ def forward(self, lhs, rhs):
 @register_test_case(module_factory=lambda: CeilFloatModule())
 def CeilFloatModule_basic(module, tu: TestUtils):
     module.forward(torch.rand(()).double(), torch.rand(()).double())
+
+
+# ==============================================================================
+
+
+class BoolFloatFalseModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([], torch.float64, True),
+    ])
+    def forward(self, a):
+        sub = float(a) - float(a)
+        return bool(torch.ops.aten.Bool(float(sub)))
+
+
+@register_test_case(module_factory=lambda: BoolFloatFalseModule())
+def BoolFloatFalseModule_basic(module, tu: TestUtils):
+    module.forward(torch.rand(()).double())
+
+
+class BoolFloatTrueModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([], torch.float64, True),
+    ])
+    def forward(self, a):
+        return bool(torch.ops.aten.Bool(float(a)))
+
+
+@register_test_case(module_factory=lambda: BoolFloatTrueModule())
+def BoolFloatTrueModule_basic(module, tu: TestUtils):
+    module.forward(torch.rand(()).double())
```

Comment:
I think `torch.rand` could in theory output `0.0`, right? I would use `tu.rand(low=0.5)`